corehq/form_processor/models.py

```diff
@@ -810,9 +810,7 @@ def to_json(self):
         ret['actions'] = lazy_serialize_case_transactions(self)
         ret['xform_ids'] = lazy_serialize_case_xform_ids(self)
         ret['case_attachments'] = lazy_serialize_case_attachments(self)
-        for key in self.case_json:
-            if key not in ret:
-                ret[key] = self.case_json[key]
+        ret = self.case_json | ret
```

Comment:
Cool new operator for dicts. In this case would it be more efficient to do this?
```py
        ret = dict(self.case_json, **serializer.data)
        ret['indices'] = lazy_serialize_case_indices(self)
        ret['actions'] = lazy_serialize_case_transactions(self)
        ret['xform_ids'] = lazy_serialize_case_xform_ids(self)
        ret['case_attachments'] = lazy_serialize_case_attachments(self)
        return ret
```
That way only one new dict is created rather than two: `ret = dict(...)` creates one and then later that is discarded and a new one created in `ret = self.case_json | ret`.

Note that `ret = dict(self.case_json, **serializer.data)` will fail if `serializer.data` has non-string keys. If that is the case, I would suggest
```py
ret = dict(self.case_json)
ret.update(serializer.data)
```
Maybe replace `.update()` with `ret |= serializer.data` if you're burning to use a new operator, but why not keep it backward-compatible?

Also note that dict ordering for all of the suggestions in this comment may differ from the existing ordering, but I'm guessing that is not a concern.

Reply:
Another option for using the operator a bit more efficiently would be to replace the first line of the above suggestion with
```py
ret = self.case_json | serializer.data
```

Reply:
+1 to `ret = self.case_json | serializer.data` (assuming that works, I seem to recall that `serializer.data` is not actually a dict)

Reply:
:+1: 03fd4bf